backend/backends/datadog/datadog.go

```diff
@@ -216,12 +216,13 @@ func (d *client) authenticatedURL(path string) string {
 
 // NewClientFromViper returns a new Datadog API client.
 func NewClientFromViper(v *viper.Viper) (backendTypes.Backend, error) {
-	v.SetDefault("datadog.timeout", defaultClientTimeout)
-	v.SetDefault("datadog.max_request_elapsed_time", defaultMaxRequestElapsedTime)
+	dd := getSubViper(v, "datadog")
+	dd.SetDefault("timeout", defaultClientTimeout)
+	dd.SetDefault("max_request_elapsed_time", defaultMaxRequestElapsedTime)
 	return NewClient(
-		v.GetString("datadog.api_key"),
-		v.GetDuration("datadog.timeout"),
-		v.GetDuration("datadog.max_request_elapsed_time"),
+		dd.GetString("api_key"),
+		dd.GetDuration("timeout"),
+		dd.GetDuration("max_request_elapsed_time"),
```

Comment:
Because of Viper bug this was set to 0 IF config file was used. And 0 means no cap on time retryer attempts to send requests. So, as soon as we hit ANY issue with sending requests, the server blocked in the infinite retry loop.